parsl/monitoring/monitoring.py

```diff
@@ -323,19 +324,24 @@ def wrapped(*args: List[Any], **kwargs: Dict[str, Any]) -> Any:
                                monitoring_hub_url,
                                run_id)
 
+            p: Optional[Process]
             if monitor_resources:
                 # create the monitor process and start
-                p: Optional[Process]
-                p = Process(target=monitor,
-                            args=(os.getpid(),
-                                  try_id,
-                                  task_id,
-                                  monitoring_hub_url,
-                                  run_id,
-                                  logging_level,
-                                  sleep_dur),
-                            name="Monitor-Wrapper-{}".format(task_id))
-                p.start()
+                pp = ForkProcess(target=monitor,
```

Comment:
Isn't the var `pp` redundant here?

Reply:
It's to do with an awkwardness in type-checking vs Optional:

p is Optional[Process], which means p might be None, which means p.start() is not valid.

Previously, `mypy` could figure out that p would never be None because it is directly assigned a Process() value, and so p.start() is safe.

With the abstraction of `ForkProcess`, `mypy` can't narrow the type of `p` down any more.

It *can* figure out that `pp` is safe to call .start() on, apparently, or perhaps just that pp is `Any`.

It's a frustrating clash of this python abstraction style vs static checking.

(an alternative here would be to put a type checking override in... I don't really have any preference)

Reply:
Alright, just wanted to understand better what's going on. This is okay.